ghost/core/core/server/services/settings/SettingsBREADService.js

```diff
@@ -229,6 +229,20 @@ class SettingsBREADService {
                 }
             }
 
+            // Detect if Stripe is now connected in live mode
+            const stripePublicKeySetting = result.find(setting => setting.get('key') === 'stripe_connect_publishable_key');
+            if (stripePublicKeySetting) {
+                const previous = stripePublicKeySetting.previousAttributes().value;
+                const current = stripePublicKeySetting.get('value');
+
+                if (current?.match(/pk_live/) && (!previous?.match(/pk_live/) || !previous)) {
+                    // Require the Stripe service here as it breaks existing tests otherwise
+                    const stripeService = require('../stripe');
```

Comment:
@allouis requiring the stripe service here, fixes the previously failing test, but I don't think this should be the right approach. 

I was looking into calling `stripeService.connect()` directly from the `settings` API endpoint (just like we do for `disconnect()`, but I don't receive the previous state and want to avoid sending this event every time when the stripe settings are edited 🤔

Reply:
@aileen Can we inject the stripe service into the constructor rather than require it here?